File: 백준/Silver/10799. 쇠막대기/쇠막대기.py
s = input().strip()

stack = 0 # 쌓여있는 쇠막대기
pieces = 0 #잘린 조각의 총 개수
i = 0 # 인덱스

while i < len(s):
  if s[i] == "(":
    #다음문자가 바로 ) 이면 레이저
    if i + 1 < len(s) and s[i+1] == ")":
      #레이저
      #현재 쌓인 막대기들을 모두 자름 
      pieces += stack
      i += 2
    else:
      stack +=1
      i +=1

  else:
    stack -=1
    pieces +=1
    i +=1
    
print(pieces)

File: 백준/Silver/10799. 쇠막대기/쇠막대기.js
const fs = require("fs");
const input = fs.readFileSync('/dev/stdin').toString().split("\n");

let s = input[0].trim()
let stack = []
let pieces = 0;


for (let i = 0; i < s.length; i++) {
  if (s[i] == "(") {
    stack.push("(")
  } else {
    // ) 닫는 거 일때,
    stack.pop()
    if (i > 0 && s[i - 1] === "(") {
      pieces += stack.length;
    } else {
      pieces += 1
    }
  }
}

console.log(pieces)